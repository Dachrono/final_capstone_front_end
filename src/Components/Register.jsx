import { useDispatch } from 'react-redux';
import { Link, useNavigate } from 'react-router-dom';
import { createUser } from '../Redux/Slices/usersSlice';

const user = (userName, nickname) => ({
  name: userName,
  username: nickname,
});

const Register = () => {
  const navigate = useNavigate();
  const dispatch = useDispatch();

  const CreateUser = (e) => {
    const btn = e.target;
    const dataForm = btn.parentElement.parentElement;
    const name = dataForm.querySelector('#name').value;
    const username = dataForm.querySelector('#username').value;
    const msg = dataForm.querySelector('#message-error');
    const response = dispatch(createUser(user(name, username)));

    response.then((data) => {
      if (typeof data.payload !== 'object') {
        msg.textContent = 'User already exist';
      } else {
        msg.textContent = '';
        navigate('/MainPage');
      }
    });
  };

  return (
    <section className="image-back d-flex-col">
      <div className="d-flex-col form-cont gap">
        <h2 className="title-form">Register</h2>
        <form className="d-flex-col gap" onSubmit={(ev) => ev.preventDefault()}>
          <span id="message-error" />
          <input id="name" type="text" placeholder="Name" className="form-input" />
          <input id="username" type="text" placeholder="Username" className="form-input" />
          <div className="btn-cont gap">
            <button type="submit" className="form-submit" onClick={(e) => CreateUser(e)}>Register</button>
            <Link to="/" className="form-submit d-flex-col">Log In</Link>
          </div>
        </form>
      </div>
    </section>
  );
};

export default Register;
