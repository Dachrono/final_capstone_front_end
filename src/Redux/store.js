import { configureStore } from '@reduxjs/toolkit';
import { logger } from 'redux-logger';
import usersReducer from './Slices/usersSlice';
import experiencesReducer from './Slices/ExperiencesSlice';
// import reservationsReducer from './ReservationsSlice/ReservationsSlice';

const store = configureStore({
  reducer: {
    users: usersReducer,
    experiences: experiencesReducer,
    // reservations: reservationsReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;
