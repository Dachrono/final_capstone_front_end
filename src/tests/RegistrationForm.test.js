import { Provider } from 'react-redux';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { render } from '@testing-library/react';
import store from '../assets/helpers/mockStore';
import Register from '../Components/Register';

test('REGISTRATION FORM', () => {
    const tree = render(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/Register']}>
          <Routes>
            <Route path="/Register" element={<Register />} />
          </Routes>
        </MemoryRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });