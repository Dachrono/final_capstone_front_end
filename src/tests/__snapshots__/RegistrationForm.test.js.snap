// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`REGISTRATION FORM 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="image-back d-flex-col"
      >
        <div
          class="d-flex-col form-cont gap"
        >
          <h2
            class="title-form"
          >
            Register
          </h2>
          <form
            class="d-flex-col gap"
          >
            <span
              id="message-error"
            />
            <input
              class="form-input"
              id="name"
              placeholder="Name"
              required=""
              type="text"
            />
            <input
              class="form-input"
              id="username"
              placeholder="Username"
              required=""
              type="text"
            />
            <div
              class="btn-cont gap"
            >
              <button
                class="form-submit"
                type="submit"
              >
                Register
              </button>
              <a
                class="form-submit d-flex-col"
                href="/"
              >
                Log In
              </a>
            </div>
          </form>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="image-back d-flex-col"
    >
      <div
        class="d-flex-col form-cont gap"
      >
        <h2
          class="title-form"
        >
          Register
        </h2>
        <form
          class="d-flex-col gap"
        >
          <span
            id="message-error"
          />
          <input
            class="form-input"
            id="name"
            placeholder="Name"
            required=""
            type="text"
          />
          <input
            class="form-input"
            id="username"
            placeholder="Username"
            required=""
            type="text"
          />
          <div
            class="btn-cont gap"
          >
            <button
              class="form-submit"
              type="submit"
            >
              Register
            </button>
            <a
              class="form-submit d-flex-col"
              href="/"
            >
              Log In
            </a>
          </div>
        </form>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
